src/vm-vcpu/src/vm.rs

```diff
@@ -58,16 +58,29 @@ pub enum Error {
 /// Dedicated [`Result`](https://doc.rust-lang.org/std/result/) type.
 pub type Result<T> = std::result::Result<T, Error>;
 
-impl KvmVm {
+/// Trait that allows the VM to signal that the VCPUs have exited.
+///
+// This trait needs Clone because each VCPU needs to be able to call the `kick` method.
+pub trait ExitHandler: Clone {
```

Comment:
The trait *itself* doesn't _need_ `Clone` afaict, but instead `Clone` needs to be trait-bounded to the `KvmVm` struct that implements it. So you can leave `ExitHandler` with no dependencies and move the bound here:
```rust
impl<EH: 'static + ExitHandler + Send + Clone> KvmVm<EH> {
```
However, since the exit handler **has** to be `Clone`able anyway because by design it's something the vmm thread and the vcpu threads share, this is more of a nit and I don't feel strongly about where the bound sits either way.

Reply:
That makes sense. I don't know which is the most idiomatic way (probably your suggestion). The reason I added it as a trait bound is because I wanted to make it clear for the implementers what constraints are needed, and it was also easier to document in the code that clone is needed because the VM is cloning the ExitHandler before sending it to the Vcpus.